### PR TITLE
feat: reorganize the describe blocks in tests and fixed typos

### PR DESCRIPTION
**Pull request by k-g-j**
## Description

Refactors the organization of the `describe` block within the tests to better reflect what is being tested and the private function calls 

Blocker for [PR 300](https://github.com/MetaMask/eth-snap-keyring/pull/300)

### CODE REVIEW

File: src/SnapKeyring.test.ts [hunk 1]
```diff
@@ -665,6 +552,131 @@ describe('SnapKeyring', () => {
       expect(await keyring.getAccounts()).toStrictEqual([]);
     });
   });
+  describe('async request redirect', () => {
```
Comment (ccharly) [inline src/SnapKeyring.test.ts:555]:
```suggestion
  
  describe('async request redirect', () => {
```

File: src/SnapKeyring.test.ts [hunk 2]
```diff
@@ -665,6 +552,131 @@ describe('SnapKeyring', () => {
...
+      });
+      const requestPromise = keyring.signPersonalMessage(
+        ethEoaAccount1.address,
+        'hello',
+      );
+
+      await expect(requestPromise).rejects.toThrow(
+        `Redirect URL domain '${origin}' is not an allowed origin by snap '${snapId}'`,
+      );
+    };
```
Comment (ccharly) [inline src/SnapKeyring.test.ts:643]: Let's move this function before the `it.each` of this scope.

Comment (k-g-j) [reply src/SnapKeyring.test.ts]: good idea! Addressed here [1de55bd](https://github.com/MetaMask/eth-snap-keyring/pull/301/commits/1de55bd86b7448457da73cd6d35d72fb805bcaac) 😄

File: src/SnapKeyring.test.ts [hunk 3]
```diff
@@ -111,497 +111,384 @@ describe('SnapKeyring', () => {
...
 
-    it('cannot change the address of an account', async () => {
-      await expect(
-        keyring.handleKeyringSnapMessage(snapId, {
-          method: KeyringEvent.AccountUpdated,
-          params: {
-            account: {
-              ...(ethEoaAccount1 as unknown as KeyringAccount),
-              address: ethEoaAccount2.address,
+      it('cannot update an account owned by another Snap', async () => {
```
Comment (ccharly) [inline src/SnapKeyring.test.ts:193]: Should probably be in the `#handleAccountCreated`

File: src/SnapKeyring.test.ts [hunk 4]
```diff
@@ -111,497 +111,384 @@ describe('SnapKeyring', () => {
...
-        ethEoaAccount2.address.toLowerCase(),
-        ethErc4337Account.address.toLowerCase(),
-        btcP2wpkhAccount.address.toLowerCase(),
-      ]);
-    });
 
-    it('returns null when removing an account that does not exist', async () => {
-      mockCallbacks.removeAccount.mockImplementation(async (address) => {
-        await keyring.removeAccount(address);
+      it('cannot update an account owned by another snap', async () => {
```
Comment (ccharly) [inline src/SnapKeyring.test.ts:222]:
```suggestion
      it('cannot update an account owned by another Snap', async () => {
```

File: src/SnapKeyring.test.ts [hunk 5]
```diff
@@ -111,497 +111,384 @@ describe('SnapKeyring', () => {
...
-        ethEoaAccount1.address,
-        'hello',
-      );
 
-      const { calls } = mockSnapController.handleRequest.mock;
-      const requestId = calls[calls.length - 1][0].request.params.id;
-      await keyring.handleKeyringSnapMessage(snapId, {
-        method: KeyringEvent.RequestRejected,
-        params: { id: requestId },
+      it('cannot delete an account owned by another snap', async () => {
```
Comment (ccharly) [inline src/SnapKeyring.test.ts:319]:
```suggestion
      it('cannot delete an account owned by another Snap', async () => {
```

File: src/SnapKeyring.test.ts [hunk 6]
```diff
@@ -111,497 +111,384 @@ describe('SnapKeyring', () => {
...
+        });
+        await expect(responsePromise).rejects.toThrow(
+          "Request 'b59b5449-5517-4622-99f2-82670cc7f3f3' not found",
         );
       });
 
-      it('throws an error if no allowed origins', async () => {
-        expect.hasAssertions();
-        await isNotAllowedOrigin([], 'https://example.com/sign?tx=1234');
+      it("cannot approve another snap's request", async () => {
```
Comment (ccharly) [inline src/SnapKeyring.test.ts:396]:
```suggestion
      it("cannot approve another Snap's request", async () => {
```

File: src/SnapKeyring.test.ts [hunk 7]
```diff
@@ -111,497 +111,384 @@ describe('SnapKeyring', () => {
...
-      );
-    });
 
-    it("fails to reject a request that doesn't exist", async () => {
-      const responsePromise = keyring.handleKeyringSnapMessage(snapId, {
-        method: KeyringEvent.RequestRejected,
-        params: {
-          id: 'b59b5449-5517-4622-99f2-82670cc7f3f3',
-        },
+      it("cannot reject another snap's request", async () => {
```
Comment (ccharly) [inline src/SnapKeyring.test.ts:468]:
```suggestion
      it("cannot reject another Snap's request", async () => {
```

File: src/SnapKeyring.test.ts [hunk 8]
```diff
@@ -666,6 +553,133 @@ describe('SnapKeyring', () => {
...
+        'https://notallowed.com/sign?tx=1234',
+      );
+    });
+
+    it('throws an error if no allowed origins', async () => {
+      expect.hasAssertions();
+      await isNotAllowedOrigin([], 'https://example.com/sign?tx=1234');
+    });
+
+    it('throws an error if the snap is undefined', async () => {
```
Comment (ccharly) [inline src/SnapKeyring.test.ts:660]:
```suggestion
    it('throws an error if the Snap is undefined', async () => {
```

**Pull request merged**